Amplify/Core/Support/Encodable+AnyEncodable.swift

```diff
@@ -0,0 +1,29 @@
+//
+// Copyright 2018-2020 Amazon.com,
+// Inc. or its affiliates. All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import Foundation
+
+public struct AnyEncodable: Encodable {
```

Comment:
1. If we're exposing this as a public utility, let's add some unit tests to make sure this works as contracted, and some docs defining its use cases.
1. We don't need a corresponding `AnyDecodable`?

Reply:
1. Done
2. `AnyDecodable` was not needed so far, so I decided to only introduce it when/if needed